RecoLocalCalo/EcalRecAlgos/src/PulseChiSqSNNLS.cc

```diff
@@ -404,6 +420,28 @@ bool PulseChiSqSNNLS::NNLS() {
   
 }
 
+void PulseChiSqSNNLS::NNLSUnconstrainParameter(Index idxp) {
+  
+  aTamat.col(_nP).swap(aTamat.col(idxp));
+  aTamat.row(_nP).swap(aTamat.row(idxp));
+  _pulsemat.col(_nP).swap(_pulsemat.col(idxp));
+  std::swap(aTbvec.coeffRef(_nP),aTbvec.coeffRef(idxp));
+  std::swap(_ampvec.coeffRef(_nP),_ampvec.coeffRef(idxp));
+  std::swap(_bxs.coeffRef(_nP),_bxs.coeffRef(idxp));
+  ++_nP;
```

Comment:
The wvec vector was actually vestigial to a previous version of the implementation and is not needed at all (this pull request in fact removes it entirely https://github.com/cms-sw/cmssw/pull/17205/files#diff-08eda0e31e35729f5f2d2af294df5134L63)